### ISSUE TITLE
Consider purging 'Cancelled' task documents immediately

### ISSUE THREAD

**Issue by dianabarsan** [medic/cht-core#6565]
Consider purging 'Cancelled' task documents immediately

**What feature do you want to improve?**
With https://github.com/medic/cht-core/issues/6181, we introduced task purging. The current rule is that tasks are purged automatically when they are in a terminal state ('Cancelled', 'Completed', 'Failed') 60 days past their due date. 

**Describe the improvement you'd like**
We could consider purging 'Cancelled' (maybe also 'Failed') task documents immediately, without waiting for them to be expired for 60 days. 
We need to explore whether this will have a negative impact, but from the obvious view, they are skipped from being indexed by the single view that is queried to get tasks on the device: 
https://github.com/medic/cht-core/blob/master/ddocs/medic-db/medic-client/views/tasks_by_contact/map.js#L3
https://github.com/medic/cht-core/blob/master/shared-libs/rules-engine/src/pouchdb-provider.js#L27

**Additional context**
If a config unadvertently creates a large number of tasks, that consequently get Cancelled when the config is updated or fixed, they will persist on the device for at least 60 days **after their due date** (so not after their cancellation date).
Couplig this with the fact that we have no upper time limit on considering emissions timely to generate tasks, if a task with a due date a year from now is Cancelled, it will only be purged one year and 60 days from now.

**Comment by garethbowen** [comment]
I don't think we want to purge them immediately because we need the Cancelled state change to replicate to the server. We could find a way to detect when they're replicated and purge them immediately after that.

I also envisage a UI improvement to Task filter or similar so you can see tasks in terminal states, though there may not be a sensible use case for viewing Cancelled tasks.

I think it would make sense to at least change the 60 days to being relative to when the Task moved to the terminal state rather than when the task was due, otherwise a task could be configured with a due date long in the future and never purge, or alternatively be purged before replication because the due date is 59 days ago.

**Comment by dianabarsan** [comment]
> I don't think we want to purge them immediately because we need the Cancelled state change to replicate to the server.
I mean purge on the server :)
